### PR TITLE
Fix error with new numpy version

### PR DESCRIPTION
**Pull request by BStoelzner**
np.product doesn't work with the new numpy, so I changed it to np.prod

### CODE REVIEW

**Comment by BStoelzner** [comment]
The tests seem to fail because of an incompatibility between numpy and pandas, see https://github.com/numpy/numpy/issues/26710
A popular solution at the moment seems to be to just downgrade numpy or pandas.

File: pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,8 @@ classifiers = [
 dynamic = ["version"]
 dependencies = [
     "deprecated",
-    "pz-rail-base",
+    "pz-rail-base>=1.0.3",
+    "numpy<2.0.0",
```
Comment (hangqianjun) [inline pyproject.toml:21]: Do we want to set the numpy version limit?

**Review by hangqianjun — COMMENTED** [review]
Looks good to me, and thanks for adding the unit tests! 

One thing I'm not sure about is the numpy version limit - I think a number of people argued against that as it might make co-updating rail and numpy harder in future versions. People also tend to use the newer versions of numpy, and reverting to older version in order to use rail may be slightly non-ideal. 

Having said that, setting `numpy<2.0.0` is an effective and short term solution, because the issue presented here seems to be unsolvable within rail, and needs actions from external packages.

Any thought from other reviewers?

**Comment by BStoelzner** [comment]
Another option would be to require `pandas>=2.2.2`, as this seems to be the first version that is compatible with numpy 2.0, see https://pandas.pydata.org/docs/whatsnew/v2.2.2.html#pandas-2-2-2-is-now-compatible-with-numpy-2-0

**Comment by BStoelzner** [comment]
I just tested setting `pandas>=2.2.2`, but this leads to a conflict with photerr because this requires `pandas<2.0.0`. @jfcrenshaw, is there a possibility to make photerr compatible with pandas 2.0?

**Comment by ztq1996** [comment]
Okay here is the thing:

photerr==1.1.0 requires pandas>1.4.3 <2.0.0
Now photerr==1.3.0 has drop the requirement to pandas>1.4.3 only, but it only supports python>3.10, so python 3.9 still won't work.

**Review by eacharles — APPROVED** [review]

**Pull request merged**